sockeye/arguments.py

```diff
@@ -1079,6 +1079,12 @@ def add_inference_args(params):
                                     "Optionally, a list of factors can be provided: "
                                     "{'text': 'some input string', 'factors': ['C C C', 'X X X']}.")
 
+    decode_params.add_argument('--input-constraints',
```

Comment:
afaik, this introduces an inconsistency with input from a file or stdin. I think we intentionally restricted constrained decoding to json input to avoid parsing multiple files and having to add tons of checks whether they have the same number of lines as the input file / stdin.

Reply:
To me it makes sense to be flexible here. You can still use the old way with json.